pkg/backend/diy/backend.go

```diff
@@ -713,6 +713,20 @@ func (b *diyBackend) CreateStack(
 		return nil, backend.ErrTeamsNotSupported
 	}
 
+	var snap *deploy.Snapshot
+	var err error
+	if initialState != nil {
+		snap, err = stack.DeserializeUntypedDeployment(ctx, initialState, stack.DefaultSecretsProvider)
+		if err != nil {
+			return nil, err
+		}
+
+		err = snap.VerifyIntegrity()
```

Comment:
I don't think we need to verify integrity here, we can assume that's already be done elsewhere. Then you don't need to use `stack.DefaultSecretsProvider` (which I've been trying to keep to just being init'd in top level commands).

Compare to ImportDeployment which just uses MarshalUntypedDeploymentToVersionedCheckpoint.

Reply:
Updated; let me know what you think :+1: